File: kfcore/src/main/java/com/lq/ren/base/ui/BaseViewModel.java
package com.lq.ren.base.ui;

import com.lq.ren.base.ui.reg.RegisterHub;

/**
 * Author lqren on 17/9/14.
 */

public abstract class BaseViewModel {

    private static final String TAG = "core.base.vm";

    private RegisterHub registerHub;

    final public void subscribeToDataStore() {
        registerHub = new RegisterHub();
        subscribeToDataStoreInternal(registerHub);
    }

    protected abstract void subscribeToDataStoreInternal(RegisterHub registerHub);

    public void unsubscribeFromDataStore() {
        if (registerHub != null) {
            registerHub.clear();
            registerHub = null;
        }
    }
}

File: kfcore/src/main/java/com/lq/ren/base/ui/BaseViewModel.kt
package com.lq.ren.base.ui

import com.lq.ren.base.ui.reg.RegisterHub

/**
 * Author lqren on 17/9/14.
 */

abstract class BaseViewModel {

    private var registerHub: RegisterHub? = null

    fun subscribeToDataStore() {
        registerHub = RegisterHub()
        subscribeToDataStoreInternal(registerHub)
    }

    protected abstract fun subscribeToDataStoreInternal(registerHub: RegisterHub)

    fun unsubscribeFromDataStore() {
        if (registerHub != null) {
            registerHub!!.clear()
            registerHub = null
        }
    }

    companion object {

        private val TAG = "core.base.vm"
    }
}
